sway/desktop/transaction.c

```diff
@@ -162,14 +178,33 @@ static void copy_container_state(struct sway_container *container,
 
 static void transaction_add_node(struct sway_transaction *transaction,
 		struct sway_node *node) {
-	struct sway_transaction_instruction *instruction =
-		calloc(1, sizeof(struct sway_transaction_instruction));
+	struct sway_transaction_instruction *instruction = NULL;
+
+	// Check if we have an instruction for this node already, in which case we
+	// update that.
+	if (node->ntxnrefs > 0) {
+		for (int idx = 0; idx < transaction->instructions->length; idx++) {
+			struct sway_transaction_instruction *other =
+				transaction->instructions->items[idx];
+			if (other->node == node) {
+				instruction = other;
+				goto update;
```

Comment:
This can be a `break`, and the code below can be within an `if (!instruction)` block, avoiding a `goto`.

Reply:
I like the goto better, but oh well. Changed.